samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -204,7 +212,15 @@ public synchronized void send(String source, OutgoingMessageEnvelope envelope) {
     checkCallbackThrowable("Received exception on message send");
 
     EventData eventData = createEventData(streamId, envelope);
-    int eventDataLength =  eventData.getBytes() == null ? 0 : eventData.getBytes().length;
+    int eventDataLength = eventData.getBytes() == null ? 0 : eventData.getBytes().length;
+    if (eventDataLength > this.maxMessageSize) {
+      LOG.info("Received a message with size {} > maxMessageSize configured {(}), Skipping it", eventDataLength,
```

Comment:
Can we also log some identifier for the message, such as the message key?

Reply:
As we discussed offline, key may contain the sensitive information. So we cannot log the key or the payload.